pkg/operator/configobserver/node/suppress_config_updates.go

```diff
@@ -110,3 +115,112 @@ func (s *revisionDiffProfileSuppressor) shouldSuppressConfigUpdates() (suppress
 	// suppress=false, when all config values are identical
 	return false, nil
 }
+
+type rejectExtremeProfilesSupressor struct {
+	operatorClient     v1helpers.StaticPodOperatorClient
+	configNodeLister   configlistersv1.NodeLister
+	latencyConfigs     []LatencyConfigProfileTuple
+	usedConfigPaths    [][]string
+	rejectionScenarios []LatencyProfileRejectionScenario
+}
+
+// NewSuppressConfigUpdateForExtremeProfilesFunc is used to create a conditional func (ShouldSuppressConfigUpdatesFunc)
+// that can be used by the latency profile config observer to determine if a new update to observedConfig should be rolled out or not.
+// It checks the current value of observedConfig from operator's state (static pod operator resource), tries to determine the latency profile
+// that was applied by the config observer and compares the same against the desired profile set on the cluster. In case, the update from
+// the active profile to the desired profile is an extreme scenario i.e. one of the scenarios specified in rejectionScenarios,
+// the suppressor suppresses the update.
+func NewSuppressConfigUpdateForExtremeProfilesFunc(
+	operatorClient v1helpers.StaticPodOperatorClient,
+	configNodeLister configlistersv1.NodeLister,
+	latencyConfigs []LatencyConfigProfileTuple,
+	rejectionScenarios []LatencyProfileRejectionScenario,
+) ShouldSuppressConfigUpdatesFunc {
+
+	usedConfigPaths := make([][]string, len(latencyConfigs))
+	for i, latencyConfig := range latencyConfigs {
+		usedConfigPaths[i] = latencyConfig.ConfigPath
+	}
+
+	ret := rejectExtremeProfilesSupressor{
+		operatorClient:     operatorClient,
+		configNodeLister:   configNodeLister,
+		latencyConfigs:     latencyConfigs,
+		usedConfigPaths:    usedConfigPaths,
+		rejectionScenarios: rejectionScenarios,
+	}
+	return ret.shouldSuppressConfigUpdates
+}
+
+func (s *rejectExtremeProfilesSupressor) shouldSuppressConfigUpdates() (bool, error) {
+	// get current state of the config
+	operatorSpec, _, _, err := s.operatorClient.GetStaticPodOperatorState()
+	if err != nil {
+		return false, err
+	}
+
+	// raw bytes of observed config in JSON format
+	observedConfigRaw, err := kyaml.ToJSON(operatorSpec.OperatorSpec.ObservedConfig.Raw)
```

Comment:
Well, initially I did do that using: `resourcemerge.MergeProcessConfig(nil, operatorSpec.OperatorSpec.ObservedConfig.Raw, operatorSpec.OperatorSpec.UnsupportedConfigOverrides.Raw)` but ended up not merging instead because, in this suppressor:
- we're checking known arg values per latency profile across the currently set observedConfig
- we're not checking against config maps (configs present in config maps are actually obtained through a merge of observed and unsupporteoverrides) 
- we're trying to determine the profile for the currently active profile from last known value of ObservedConfig directly
- if we merge ObservedConfig + UnsupportedConfigOverrides in our current approach, comparison would be faulty as `fromProfile` wouldn't be able to determined from the `knownProfileConfigs` and we'd end up returning some error / degrading the `configobservationcontroller` whenever this suppressor returns that error